harness/determined/pytorch/_pytorch_context.py

```diff
@@ -572,7 +580,7 @@ def backward(
                         gradient=gradient, retain_graph=retain_graph, create_graph=create_graph
                     )
 
-                if self.distributed.size > 1 and self._should_communicate_and_update():
+                if self.distributed.size > 1 and self._should_communicate_and_update() and self._distributed_backend.use_horovod():
                     # When we exit out of this context manager, we need to finish
                     # communicating gradient updates before they are unscaled.
                     #
```

Comment:
github won't let me comment on lines that are distant from changed lines, but I think you have a fairly serious networking inefficiency by not using [DDP.no_sync](https://pytorch.org/docs/master/generated/torch.nn.parallel.DistributedDataParallel.html#torch.nn.parallel.DistributedDataParallel.no_sync) during gradient aggregation.

Reply:
introduced no_sync.